### PR TITLE
Refract - handle complex array members

### PR DESCRIPTION
**Pull request by klokane**
- handle exception on test/fixtures/render-array-nested-object.apib

### CODE REVIEW

File: src/RefractAST.cc [hunk 1]
```diff
@@ -570,14 +573,31 @@ namespace drafter
...
+                return RefractElementFromValue<refract::ArrayElement>(value);
+
             case mson::ObjectTypeName :
+                return RefractElementFromValue<refract::ObjectElement>(value);
+
+            case mson::UndefinedTypeName:
+                if(ValueHasChildren(value)) {
+                    return RefractElementFromValue<refract::ArrayElement>(value);
+                }
+                else if(!value.valueDefinition.typeDefinition.typeSpecification.name.symbol.literal.empty() || ValueHasMembers(value)) {
```
Comment (pksunkara) [inline src/RefractAST.cc]: Please divide the condition into two lines. It's really long.

```
else if (!value.valueDefinition.typeDefinition.typeSpecification.name.symbol.literal.empty() ||
         ValueHasMembers(value)) {
```

File: src/RefractAST.cc [hunk 2]
```diff
@@ -570,14 +573,31 @@ namespace drafter
...
+            }
+
             case mson::ArrayTypeName :
+                return RefractElementFromValue<refract::ArrayElement>(value);
+
             case mson::ObjectTypeName :
+                return RefractElementFromValue<refract::ObjectElement>(value);
+
+            case mson::UndefinedTypeName:
+                if(ValueHasChildren(value)) {
```
Comment (pksunkara) [inline src/RefractAST.cc]: `if (`

**Comment by pksunkara** [comment]
Reviewed. Will merge once I check the tests on my computer.

**Comment by pksunkara** [comment]
I just corrected the tests in my branch and rebased this branch over that. There is still this [example](https://github.com/apiaryio/drafter/blob/fc0af715be47d7757aa2713d25af1f8ba2bd1f78/test/fixtures/render-array-ref-array.apib) whose type expansion needs to be fixed.

**Comment by klokane** [comment]
replaced by #58

**Pull request closed without merging**